src/main/kotlin/org/rust/lang/core/completion/RsDeriveCompletionProvider.kt

```diff
@@ -5,25 +5,25 @@ import com.intellij.codeInsight.completion.CompletionProvider
 import com.intellij.codeInsight.completion.CompletionResultSet
 import com.intellij.codeInsight.lookup.LookupElementBuilder
 import com.intellij.patterns.ElementPattern
-import com.intellij.patterns.ElementPatternCondition
 import com.intellij.patterns.PatternCondition
 import com.intellij.patterns.PlatformPatterns.psiElement
 import com.intellij.psi.PsiElement
 import com.intellij.util.ProcessingContext
 import org.rust.lang.RsLanguage
-import org.rust.lang.core.psi.RsOuterAttr
 import org.rust.lang.core.psi.RsElementTypes.*
+import org.rust.lang.core.psi.RsOuterAttr
 import org.rust.lang.core.psi.util.parentOfType
 
-object DeriveCompletionProvider : CompletionProvider<CompletionParameters>() {
+object RsDeriveCompletionProvider : CompletionProvider<CompletionParameters>() {
 
     private val DERIVABLE_TRAITS = listOf(
         "Eq", "PartialEq",
         "Ord", "PartialOrd",
         "Copy", "Clone",
         "Hash",
         "Default",
-        "Debug"
+        "Debug",
+        "RustcEncodable", "RustcDecodable"
```

Comment:
@matklad, but you still have to attach `rustc_serializable` to be able to use `RustcEncodable`.

In this "don't show completions we can't use right away" / "let the user complete what he wants and resolve dependency problems later" dilemma, I have a slight inclination to the latter. I think we shouldn't prevent completion just because the user hasn't added the dependency yet. If he or she has already decided to make a struct De/Encodable, they eventually add the dependency. But that only relates to the standard derivables the compiler knows of. This approach can't be spread to third party derivables like `(De)Serialize` from serde.